### ISSUE TITLE
Feature: Support row and cell-level style overrides

### ISSUE THREAD

**Issue by erhhung** [petercorke/ansitable#10]
Introduce `Row` and `Cell` classes to allow customizing per-row and per-cell styling (alignment, color, and style). This is useful, for example, to **highlight specific rows or specific cell values** in a large table using, e.g., `style="reverse"`

```py
# proposed syntax only!
table.add(Row(["hello-world", "1.0.0", "2023-07-28"], style="reverse")) #  row-level override
table.add(Row([Cell("foo-bar", style="bold"), "0.1.0", "2023-07-28"]))  # cell-level override
```

**Comment by petercorke** [comment]
thanks for the various issues and sorry for being slow,  `ansitable` is probably due some TLC.

The `Row` and `Cell` ideas are interesting.  At the moment the cell styling is done is a weak way with magic text.  What does `Row` give beyond adding style arguments to `table.add`?

**Comment by petercorke** [comment]
If you've fixed any of these issues please share a PR.  They are good points, and hopefully I can get some time to get back into this before end of year.

**Comment by petercorke** [comment]
OK, the row level override now exists, trailing arguments passed to `row()`.

Am thinking of a few mechanisms to highlight specific cells in a row, perhaps something like:

```
table.row(*list, highlight=[Cell(0, bgcolor="red"), Cell(1, style="bold")])
```

which would require a new class `Cell`.

**Comment by erhhung** [comment]
I think separating the individual cell values from another `highlight` array that provides cell overrides is not very clean.

A common use case is to highlight an entire row, meaning the `color` and `bgcolor` is consistent across all cells of that row. So if I'm calling `table.row(*list)`, I'd expect an intuitive list of optional args (e.g. `color` and `bgcolor`) that would apply the same style all cells of that row.

If I wanted to override the styling of specific cells in a row, then calling `table.row(Cell("foo", color="red"), "bar", Cell("baz", bgcolor="yellow", style="underlined"))` would be more appropriate, IMO.

**Comment by petercorke** [comment]
The common use case is now implemented, you can now set `fgcolor`, `bgcolor` and `style` for a row that override columnise settings.

Your syntax is elegant, thanks for the suggestion.  I will go that way.

**Comment by petercorke** [comment]
This has been done, it's included in 0.11.2 which is current on PyPI.  There's also Sphinx doco now at https://petercorke.github.io/ansitable/
